contract/src/main/java/com/ritense/valtimo/contract/json/patch/JsonPatchBuilder.java

```diff
@@ -93,28 +93,45 @@ public JsonPatchBuilder addJsonNodeValue(JsonNode destination, JsonPointer path,
     }
 
     /**
-     * This will adjust the first unindexed array position ('/-') to a new index depending on previous operations or destination object.
+     * This will adjust the first un-indexed array position ('/-' or '/+') to an index depending on previous operations or destination object.
+     * The '/-' will add the JSON to a new element in the array.
+     * The '/+' will append JSON to the last element in the array, or create a new element.
      * Any subsequent occurrences of '/-' will be replaced by 0, as the first one will create a new node already.
-     *
      * <p>Example (where x in destination has a length of 1):
      * /x/-/y/-/z -> /x/1/y/0/z</p>
      */
     private JsonPointer determineUnindexedPath(JsonNode destination, JsonPointer path) {
         String stringPath = path.toString();
         int dashIndex = stringPath.indexOf("/-");
-        if (dashIndex == -1) {
+        int plusIndex = stringPath.indexOf("/+");
+        if (dashIndex == -1 && plusIndex == -1) {
             return path;
-        }
-
-        String arrayPath = stringPath.substring(0, dashIndex);
-        for (int i = 0; ; i++) {
-            String testPath = arrayPath + "/" + i;
-            if (operations.stream().noneMatch(op -> op.getPath().equals(testPath))
-                && destination.at(testPath).isMissingNode()
-            ) {
-                String correctedPath = testPath + stringPath.substring(dashIndex + 2)
-                    .replace("/-", "/0");
-                return JsonPointer.compile(correctedPath);
+        } else if (dashIndex != -1) {
+            String arrayPath = stringPath.substring(0, dashIndex);
+            for (int i = 0; ; i++) {
+                String testPath = arrayPath + "/" + i;
+                if (operations.stream().noneMatch(op -> op.getPath().equals(testPath))
+                    && destination.at(testPath).isMissingNode()
+                ) {
+                    String correctedPath = testPath + stringPath.substring(dashIndex + 2)
+                        .replace("/-", "/0");
+                    return JsonPointer.compile(correctedPath);
+                }
+            }
+        } else {
```

Comment:
You're missing the scenario where there is neither a - nor a + in the string. What happens then?

Reply:
Line 107 will handle when no - or + is in the string